pcsx2/GS/Renderers/HW/GSTextureCache.cpp

```diff
@@ -1000,6 +1000,12 @@ void GSTextureCache::InvalidateVideoMem(const GSOffset& off, const GSVector4i& r
 						continue;
 					}
 				}
+				else if (GSConfig.UserHacks_TextureInsideRt && t->Overlaps(bp, bw, psm, rect) && GSUtil::HasCompatibleBits(psm, t->m_TEX0.PSM))
```

Comment:
I'm pretty sure there's a game (Jak X?) that hits the previous branch but it fails to recognize the overlap correctly, just for reference.